### ISSUE TITLE
Timeline variable order randomisation is not turned off by a value of `false`

### ISSUE THREAD

**Issue by shirdekel** [djnavarro/jaysire#24]
Applying `set_parameters(randomize_order = FALSE)` to a timeline with timeline variables does not turn off randomisation of those variables. However, it does turn off when re-rewriting the JS code in a way that is more similar to the jsPsych demos.

# jaysire code

``` r
library(jaysire)

trial_template <- trial_html_button_response(
  stimulus = insert_variable("stimulus"),
  choices = c("true", "false")
)

numbers <- c(
  1:5
)

trials <- build_timeline(trial_template) %>%
  set_variables(stimulus = numbers) %>%
  set_parameters(randomize_order = FALSE)


build_experiment(
  timeline = build_timeline(trials),
  path = temporary_folder()
)
#> Warning in dir.create(path): '/var/folders/5t/6fk0dcqn7vv1_8ln0fq3dvp00000gp/T//
#> Rtmp7viiFd/jaysire_dcplz' already exists
```

<sup>Created on 2020-08-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 4.0.2 (2020-06-22)
#>  os       macOS Mojave 10.14.6        
#>  system   x86_64, darwin17.0          
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_AU.UTF-8                 
#>  ctype    en_AU.UTF-8                 
#>  tz       Australia/Sydney            
#>  date     2020-08-26                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date       lib source                            
#>  assertthat    0.2.1   2019-03-21 [1] CRAN (R 4.0.0)                    
#>  backports     1.1.9   2020-08-24 [1] CRAN (R 4.0.2)                    
#>  callr         3.4.3   2020-03-28 [1] CRAN (R 4.0.0)                    
#>  cli           2.0.2   2020-02-28 [1] CRAN (R 4.0.0)                    
#>  crayon        1.3.4   2017-09-16 [1] CRAN (R 4.0.0)                    
#>  desc          1.2.0   2018-05-01 [1] CRAN (R 4.0.0)                    
#>  devtools      2.3.0   2020-04-10 [1] CRAN (R 4.0.0)                    
#>  digest        0.6.25  2020-02-23 [1] CRAN (R 4.0.0)                    
#>  ellipsis      0.3.1   2020-05-15 [1] CRAN (R 4.0.0)                    
#>  evaluate      0.14    2019-05-28 [1] CRAN (R 4.0.0)                    
#>  fansi         0.4.1   2020-01-08 [1] CRAN (R 4.0.0)                    
#>  fs            1.5.0   2020-07-31 [1] CRAN (R 4.0.2)                    
#>  glue          1.4.1   2020-05-13 [1] CRAN (R 4.0.0)                    
#>  highr         0.8     2019-03-20 [1] CRAN (R 4.0.0)                    
#>  htmltools     0.5.0   2020-06-16 [1] CRAN (R 4.0.1)                    
#>  jaysire     * 0.1.0   2020-08-26 [1] Github (djnavarro/jaysire@5fa934d)
#>  jsonlite      1.7.0   2020-06-25 [1] CRAN (R 4.0.0)                    
#>  knitr         1.29    2020-06-23 [1] CRAN (R 4.0.2)                    
#>  magrittr      1.5     2014-11-22 [1] CRAN (R 4.0.0)                    
#>  memoise       1.1.0   2017-04-21 [1] CRAN (R 4.0.0)                    
#>  pkgbuild      1.1.0   2020-07-13 [1] CRAN (R 4.0.2)                    
#>  pkgload       1.1.0   2020-05-29 [1] CRAN (R 4.0.0)                    
#>  prettyunits   1.1.1   2020-01-24 [1] CRAN (R 4.0.0)                    
#>  processx      3.4.3   2020-07-05 [1] CRAN (R 4.0.2)                    
#>  ps            1.3.4   2020-08-11 [1] CRAN (R 4.0.2)                    
#>  purrr         0.3.4   2020-04-17 [1] CRAN (R 4.0.0)                    
#>  R6            2.4.1   2019-11-12 [1] CRAN (R 4.0.0)                    
#>  remotes       2.1.1   2020-02-15 [1] CRAN (R 4.0.0)                    
#>  rlang         0.4.7   2020-07-09 [1] CRAN (R 4.0.1)                    
#>  rmarkdown     2.3     2020-06-18 [1] CRAN (R 4.0.1)                    
#>  rprojroot     1.3-2   2018-01-03 [1] CRAN (R 4.0.0)                    
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.0.0)                    
#>  stringi       1.4.6   2020-02-17 [1] CRAN (R 4.0.0)                    
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.0.0)                    
#>  testthat      2.3.2   2020-03-02 [1] CRAN (R 4.0.0)                    
#>  usethis       1.6.1   2020-04-29 [1] CRAN (R 4.0.0)                    
#>  withr         2.2.0   2020-04-20 [1] CRAN (R 4.0.0)                    
#>  xfun          0.16    2020-07-24 [1] CRAN (R 4.0.2)                    
#>  yaml          2.2.1   2020-02-01 [1] CRAN (R 4.0.0)                    
#> 
#> [1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library
```

</details>

# Resulting JS

```
var timeline = {
  "timeline": [
    {
      "timeline": [
        {
          "type": ["html-button-response"],
          "stimulus": jsPsych.timelineVariable('stimulus'),
          "choices": ["true", "false"],
          "margin_vertical": ["0px"],
          "margin_horizontal": ["8px"],
          "response_ends_trial": true,
          "post_trial_gap": [1000]
        }
      ],
      "timeline_variables": [
        {
          "stimulus": [1]
        },
        {
          "stimulus": [2]
        },
        {
          "stimulus": [3]
        },
        {
          "stimulus": [4]
        },
        {
          "stimulus": [5]
        }
      ],
      "randomize_order": [false],
      "repetitions": [1]
    }
  ]
};

jsPsych.init(
{
  "timeline": [timeline]
}
);
```

The numbers are still displayed randomly.

# Re-writing

```
var timeline = [];

var numbers = [
    {
      "stimulus": [1]
    },
    {
      "stimulus": [2]
    },
    {
      "stimulus": [3]
    },
    {
      "stimulus": [4]
    },
    {
      "stimulus": [5]
    }
];

var trials = {
  type: 'html-button-response',
  stimulus: jsPsych.timelineVariable('stimulus'),
  choices: ["true", "false"],
  margin_vertical: "0px",
  margin_horizontal: "8px",
  response_ends_trial: true,
  post_trial_gap: 1000
  }

var test_procedure = {
  timeline: [trials],
  timeline_variables: numbers,
  randomize_order: false,
  repetitions: 1
}
timeline.push(test_procedure);

jsPsych.init({
  timeline: timeline
});
```

The numbers are now displayed in order.

**Comment by shirdekel** [comment]
It seems as though the `false` vs `[false]` difference is the culprit.

**Comment by djnavarro** [comment]
Thanks!

**Comment by shirdekel** [comment]
Would you like to review a pull request? I don't know if it's ideal, but it works for me.

**Comment by djnavarro** [comment]
Yes please. Might not be able to look at it today but I would appreciate it!

**Comment by djnavarro** [comment]
closed by #25
